### ISSUE TITLE
ALX: a program that adds positive numbers.

### ISSUE THREAD

**Issue by ahmed149** [ahmed149/learning-programing-in-C#63]
#include <stdio.h>
#include <stdlib.h>

/**
 * main - Prints the addition of positive numbers,
 *        followed by a new line.
 * @argc: The number of arguments passed to the program.
 * @argv: An array of pointers to the arguments.
 *
 * Return: If one of the numbers contains symbols that are non-digits - 1.
 *         Otherwise - 0.
 */
int main(int argc, char *argv[])
{
	int num, digit, sum = 0;

	for (num = 1; num < argc; num++)
	{
		for (digit = 0; argv[num][digit]; digit++)
		{
			if (argv[num][digit] < '0' || argv[num][digit] > '9')
			{
				printf("Error\n");
				return (1);
			}
		}

		sum += atoi(argv[num]);
	}

	printf("%d\n", sum);

	return (0);
}

**Comment by ahmed149** [comment]
gcc -Wall -pedantic -Werror -Wextra -std=gnu89 4-add.c -o add

**Comment by ahmed149** [comment]
Certainly! I'll explain the program step by step without providing code and offer additional examples as requested.

This program is a C program that calculates the sum of positive numbers passed as command-line arguments. Here's a breakdown of what's happening:

1. `int main(int argc, char *argv[])`:
   - `argc` stands for "argument count," and it tells you how many arguments (pieces of information) were given to the program when you run it.
   - `argv` stands for "argument vector" and is an array of strings (text) that contains those arguments.

2. `for (num = 1; num < argc; num++)`:
   - This loop iterates through the command-line arguments starting from the second argument (index 1) because the first argument (`argv[0]`) is typically the program name.

3. `for (digit = 0; argv[num][digit]; digit++)`:
   - This nested loop goes through each character in the current argument (`argv[num]`) to check if all the characters are digits.

4. `if (argv[num][digit] < '0' || argv[num][digit] > '9')`:
   - Inside the inner loop, it checks whether the current character is not a digit. If it finds a non-digit character, it prints an error message and returns 1 to indicate an error.

5. `sum += atoi(argv[num]);`:
   - If all characters in the current argument are digits, it converts the argument to an integer using `atoi` and adds it to the running total `sum`.

6. `printf("%d\n", sum);`:
   - Finally, it prints the sum of all valid positive numbers.

So, to summarize:
- The program checks each command-line argument one by one.
- For each argument, it checks if all the characters are digits.
- If any argument contains non-digit characters, it prints "Error" and returns 1.
- If all arguments are valid positive numbers, it calculates their sum and prints it.

For example:
- If you run the program with `./program 5 7 10`, it will print `22`, which is the result of adding 5, 7, and 10.
- If you run it with `./program 5 abc 10`, it will print "Error" and return 1 because "abc" is not a valid positive number.

I hope this explanation helps! If you have more questions or need further examples, feel free to ask.
